core/common/src/main/java/alluxio/Seekable.java

```diff
@@ -26,4 +26,9 @@
    * @param pos the position to seek to, it must be between 0 and the end of the stream - 1
    */
   void seek(long pos) throws IOException;
+
+  /**
+   * @return the current position of the stream
+   */
+  long getPos();
```

Comment:
`Positioned` could be useful for a stream that can report its position but doesn't support seek. We could have `Seekable extends Positioned` if we think all `Seekable`s should be able to report position. Or we can leave the methods fully bundled - I was thinking more as a Gopher than Java bean when making my suggestion.